src/Mod/Arch/ArchWindowPresets.py

```diff
@@ -51,6 +51,11 @@ def makeSketch(windowtype,width,height,h1,h2,h3,w1,w2,o1,o2):
         w2 = float(w2)
         o1 = float(o1)
         o2 = float(o2)
+        # h1, h2, w1, w2 cannot be null (for now)
+        # TODO allow these to be null (don't create the component if so)
+        if not h1 or not h2 or not w1 or not w2:
+            FreeCAD.Console.PrintError("H1, H2, W1 and W2 parameters cannot be zero. Aborting\n")
+            return
```

Comment:
Why not something just like
```
if h1*h2*w1*w2 == 0:
    FreeCAD.Console.PrintError("H1, H2, W1 and W2 parameters cannot be zero. Aborting\n")
    return
```
at the very beginning of the function ?

Reply:
Indeed that's more elegant. But in any case, this test should vanish in the future (see the TODO comment above)

Reply:
```suggestion
        # h1, h2, w1, w2 cannot be null (for now)
        # TODO allow these to be null (don't create the component if so)
        if h1*h2*w1*w2 == 0:
            FreeCAD.Console.PrintError("H1, H2, W1 and W2 parameters cannot be zero. Aborting\n")
            return
```